### PR TITLE
aws_efs_file_system: Support single Availability Zone storage classes

### PR DESCRIPTION
**Pull request by shuheiktgw**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes https://github.com/hashicorp/terraform-provider-aws/issues/18016

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccAWSEFSFileSystem_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSEFSFileSystem_ -timeout 180m
=== RUN   TestAccAWSEFSFileSystem_basic
=== PAUSE TestAccAWSEFSFileSystem_basic
=== RUN   TestAccAWSEFSFileSystem_availabilityZoneName
=== PAUSE TestAccAWSEFSFileSystem_availabilityZoneName
=== RUN   TestAccAWSEFSFileSystem_tags
=== PAUSE TestAccAWSEFSFileSystem_tags
=== RUN   TestAccAWSEFSFileSystem_pagedTags
=== PAUSE TestAccAWSEFSFileSystem_pagedTags
=== RUN   TestAccAWSEFSFileSystem_kmsKey
=== PAUSE TestAccAWSEFSFileSystem_kmsKey
=== RUN   TestAccAWSEFSFileSystem_kmsConfigurationWithoutEncryption
=== PAUSE TestAccAWSEFSFileSystem_kmsConfigurationWithoutEncryption
=== RUN   TestAccAWSEFSFileSystem_ProvisionedThroughputInMibps
=== PAUSE TestAccAWSEFSFileSystem_ProvisionedThroughputInMibps
=== RUN   TestAccAWSEFSFileSystem_ThroughputMode
=== PAUSE TestAccAWSEFSFileSystem_ThroughputMode
=== RUN   TestAccAWSEFSFileSystem_lifecyclePolicy
=== PAUSE TestAccAWSEFSFileSystem_lifecyclePolicy
=== RUN   TestAccAWSEFSFileSystem_lifecyclePolicy_update
=== PAUSE TestAccAWSEFSFileSystem_lifecyclePolicy_update
=== RUN   TestAccAWSEFSFileSystem_lifecyclePolicy_removal
=== PAUSE TestAccAWSEFSFileSystem_lifecyclePolicy_removal
=== RUN   TestAccAWSEFSFileSystem_disappears
=== PAUSE TestAccAWSEFSFileSystem_disappears
=== CONT  TestAccAWSEFSFileSystem_basic
=== CONT  TestAccAWSEFSFileSystem_ThroughputMode
=== CONT  TestAccAWSEFSFileSystem_disappears
=== CONT  TestAccAWSEFSFileSystem_lifecyclePolicy_removal
=== CONT  TestAccAWSEFSFileSystem_lifecyclePolicy_update
=== CONT  TestAccAWSEFSFileSystem_lifecyclePolicy
=== CONT  TestAccAWSEFSFileSystem_kmsKey
=== CONT  TestAccAWSEFSFileSystem_ProvisionedThroughputInMibps
=== CONT  TestAccAWSEFSFileSystem_kmsConfigurationWithoutEncryption
=== CONT  TestAccAWSEFSFileSystem_tags
=== CONT  TestAccAWSEFSFileSystem_availabilityZoneName
=== CONT  TestAccAWSEFSFileSystem_pagedTags
--- PASS: TestAccAWSEFSFileSystem_kmsConfigurationWithoutEncryption (43.36s)
--- PASS: TestAccAWSEFSFileSystem_disappears (73.41s)
--- PASS: TestAccAWSEFSFileSystem_availabilityZoneName (109.55s)
--- PASS: TestAccAWSEFSFileSystem_pagedTags (123.93s)
--- PASS: TestAccAWSEFSFileSystem_kmsKey (132.33s)
--- PASS: TestAccAWSEFSFileSystem_lifecyclePolicy (136.02s)
--- PASS: TestAccAWSEFSFileSystem_ThroughputMode (147.03s)
--- PASS: TestAccAWSEFSFileSystem_lifecyclePolicy_update (151.07s)
--- PASS: TestAccAWSEFSFileSystem_lifecyclePolicy_removal (156.76s)
--- PASS: TestAccAWSEFSFileSystem_basic (157.14s)
--- PASS: TestAccAWSEFSFileSystem_ProvisionedThroughputInMibps (172.96s)
--- PASS: TestAccAWSEFSFileSystem_tags (196.84s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	198.815s

$ make testacc TESTARGS='-run=TestAccDataSourceAwsEfsFileSystem_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccDataSourceAwsEfsFileSystem_ -timeout 180m
=== RUN   TestAccDataSourceAwsEfsFileSystem_id
=== PAUSE TestAccDataSourceAwsEfsFileSystem_id
=== RUN   TestAccDataSourceAwsEfsFileSystem_name
=== PAUSE TestAccDataSourceAwsEfsFileSystem_name
=== RUN   TestAccDataSourceAwsEfsFileSystem_availabilityZone
=== PAUSE TestAccDataSourceAwsEfsFileSystem_availabilityZone
=== RUN   TestAccDataSourceAwsEfsFileSystem_NonExistent
=== PAUSE TestAccDataSourceAwsEfsFileSystem_NonExistent
=== CONT  TestAccDataSourceAwsEfsFileSystem_id
=== CONT  TestAccDataSourceAwsEfsFileSystem_name
=== CONT  TestAccDataSourceAwsEfsFileSystem_NonExistent
=== CONT  TestAccDataSourceAwsEfsFileSystem_availabilityZone
--- PASS: TestAccDataSourceAwsEfsFileSystem_NonExistent (11.69s)
--- PASS: TestAccDataSourceAwsEfsFileSystem_id (45.53s)
--- PASS: TestAccDataSourceAwsEfsFileSystem_name (45.54s)
--- PASS: TestAccDataSourceAwsEfsFileSystem_availabilityZone (45.91s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	47.897s
```

### CODE REVIEW

File: aws/data_source_aws_efs_file_system.go [hunk 1]
```diff
@@ -22,6 +22,14 @@ func dataSourceAwsEfsFileSystem() *schema.Resource {
 				Type:     schema.TypeString,
 				Computed: true,
 			},
+			"availability_zone_id": {
```
Comment (bflad) [inline aws/data_source_aws_efs_file_system.go:25]: Hi @shuheiktgw 👋  Can you please add changelog entries for these new schema attributes? Reference: https://github.com/hashicorp/terraform-provider-aws/blob/main/docs/contributing/pullrequest-submission-and-lifecycle.md#changelog-process

e.g. a new file called `.changelog/18319.txt` with contents:

``````markdown
```release-note:enhancement
data-source/aws_efs_file_system: Add `availability_zone_id` and `availability_zone_name` attributes
```

```release-note:enhancement
resource/aws_efs_file_system: Add `availability_zone_id` attribute and `availability_zone_name` argument
```
``````

Comment (shuheiktgw) [reply aws/data_source_aws_efs_file_system.go:25]: Sorry that I haven't noticed the new change log format! I'll fix it now!

**Comment by shuheiktgw** [comment]
@bflad I've added change logs and resolved the confilict. Would you review the PR again?

**Review by bflad — APPROVED** [review]
Looks great, thank you, @shuheiktgw 🚀 

Output from acceptance testing in AWS Commercial:

```
--- PASS: TestAccAWSEFSFileSystem_availabilityZoneName (52.85s)
--- PASS: TestAccAWSEFSFileSystem_basic (100.05s)
--- PASS: TestAccAWSEFSFileSystem_disappears (89.46s)
--- PASS: TestAccAWSEFSFileSystem_kmsConfigurationWithoutEncryption (11.16s)
--- PASS: TestAccAWSEFSFileSystem_kmsKey (78.45s)
--- PASS: TestAccAWSEFSFileSystem_lifecyclePolicy (135.47s)
--- PASS: TestAccAWSEFSFileSystem_lifecyclePolicy_removal (151.08s)
--- PASS: TestAccAWSEFSFileSystem_lifecyclePolicy_update (86.82s)
--- PASS: TestAccAWSEFSFileSystem_pagedTags (50.41s)
--- PASS: TestAccAWSEFSFileSystem_ProvisionedThroughputInMibps (100.30s)
--- PASS: TestAccAWSEFSFileSystem_tags (115.88s)
--- PASS: TestAccAWSEFSFileSystem_ThroughputMode (159.30s)

--- PASS: TestAccDataSourceAwsEfsFileSystem_availabilityZone (65.70s)
--- PASS: TestAccDataSourceAwsEfsFileSystem_id (123.56s)
--- PASS: TestAccDataSourceAwsEfsFileSystem_name (133.45s)
--- PASS: TestAccDataSourceAwsEfsFileSystem_NonExistent (5.64s)
```

Output from acceptance testing in AWS GovCloud (US):

```
--- PASS: TestAccAWSEFSFileSystem_availabilityZoneName (152.34s)
--- PASS: TestAccAWSEFSFileSystem_basic (126.80s)
--- PASS: TestAccAWSEFSFileSystem_disappears (215.97s)
--- PASS: TestAccAWSEFSFileSystem_kmsConfigurationWithoutEncryption (14.42s)
--- PASS: TestAccAWSEFSFileSystem_kmsKey (83.93s)
--- PASS: TestAccAWSEFSFileSystem_lifecyclePolicy (120.63s)
--- PASS: TestAccAWSEFSFileSystem_lifecyclePolicy_removal (146.67s)
--- PASS: TestAccAWSEFSFileSystem_lifecyclePolicy_update (116.64s)
--- PASS: TestAccAWSEFSFileSystem_pagedTags (144.91s)
--- PASS: TestAccAWSEFSFileSystem_ProvisionedThroughputInMibps (106.71s)
--- PASS: TestAccAWSEFSFileSystem_tags (149.15s)
--- PASS: TestAccAWSEFSFileSystem_ThroughputMode (97.13s)

--- PASS: TestAccDataSourceAwsEfsFileSystem_availabilityZone (147.25s)
--- PASS: TestAccDataSourceAwsEfsFileSystem_id (70.23s)
--- PASS: TestAccDataSourceAwsEfsFileSystem_name (110.46s)
--- PASS: TestAccDataSourceAwsEfsFileSystem_NonExistent (7.44s)
```

**Pull request merged**